crates/wascc-provider/src/states/running.rs

```diff
@@ -0,0 +1,26 @@
+use kubelet::state::{State, Transition};
+use kubelet::{
+    pod::{Phase, Pod},
+    state,
+};
+
+use crate::{make_status, PodState};
+
+use super::error::Error;
+use super::finished::Finished;
+
+state!(
+    /// The Kubelet is running the Pod.
+    Running,
+    PodState,
+    Finished,
+    Error,
+    {
+        // Wascc has no notion of exiting so we just sleep.
+        // I _think_ that periodically awaiting will allow the task to be interrupted.
```

Comment:
I think you are right here. Anything that is awaiting can be parked. But I don't know all of the implementation details under the hood

Reply:
I'm hoping that as long as we tell our providers to follow the rule of thumb of never running blocking code in their async handlers, that we don't have to worry too much about zombie workloads.